src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionEndpoint.cs

```diff
@@ -147,6 +147,13 @@ internal async Task<RazorCodeActionContext> GenerateRazorCodeActionContextAsync(
                 request.Range = request.Context.SelectionRange;
             }
 
+            // Can occur with certain race conditions when the request and
+            // current source text differ. We can return early.
```

Comment:
Wait really how? This sounds pretty bad

Reply:
My hypothesis here is some (major) change in the document occurred between code action being requested by VS and us processing/providing the code action. 

Haven't been able to get any consistent repro, however @allisonchou and I have both encountered it. Without some more information, I don't think this will be root cause'able. For now, I think this may be a good guard to add.

Reply:
Will our new logging infrastructure aid in diagnosing this?

Reply:
Unfortunately, I don't think logging will be of much use either. In this kind of situation, the suspected change in the document could occur at any point (and likely "long" before we get the sourceText on line 135). We could log this event, but not too sure if it would help clarify anything.

Reply:
I was able to get a repro pretty consistently with these steps:
1) Launch aspnetcore-tooling repo instance in debug mode
2) Open a Razor sln
3) Quickly type an '@' character at the end of the file before the sln fully loads
4) Exception appears